include/convex_bodies/hpolytope.h

```diff
@@ -490,6 +497,197 @@ class HPolytope{
 
     void free_them_all() {}
 
+#ifndef DISABLE_NLP_ORACLES
+
+    template <class bfunc>
+    std::tuple<NT, Point, int> curve_intersect(NT t_prev, NT t0, NT eta, std::vector<Point> &coeffs, bfunc phi, bfunc grad_phi, const std::string method="newton-raphson") {
+        if (method == "newton-raphson") {
```

Comment:
Could you please locate a specific file and line to adjust the design pattern accordingly? 

AFAIU, you mean a pointer to a member function (according to which we do the `curve_intersect`). [See here also](https://stackoverflow.com/questions/8865766/get-a-pointer-to-objects-member-function)

Reply:
I have this example in mind. The volume algorithm as instantiated here: https://github.com/GeomScale/volume_approximation/blob/develop/test/volume_cb_hpolytope.cpp#L65 is using the `CDHRWalk` defined here https://github.com/GeomScale/volume_approximation/blob/develop/include/random_walks/uniform_cdhr_walk.hpp#L23
Of course you may want to pass a reference to an object e.g. NewtonRaphson that has an apply method to do the computation.

Reply:
Eventually, I have kept the non-linear boundaries as members of the polytope classes and added the macro `DISABLE_NLP_ORACLES` to exclude them from the compilation, if desired. 
Another thing I had in mind when I decided to use the macros is keeping the architecture consistent since
`line_intersect_*` methods are member functions. 
The helper functions can indeed be called using their pointers but I do not see any point in doing this. 

Shall I move everything out, eventually, and do as you propose?

Reply:
NLP oracles refactored.